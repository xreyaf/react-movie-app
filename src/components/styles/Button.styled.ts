import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const ButtonStyled = styled(motion.button)`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0.5rem;
  outline: none;
  border: none;
  border-radius: 1rem;
  background: ${theme.colors.primary400};
  color: ${theme.colors.white100};
  padding: 1rem 2rem;
  font-size: 1rem;
  margin-bottom: .6rem;
  svg{
    align-self: baseline;
  }
`;
