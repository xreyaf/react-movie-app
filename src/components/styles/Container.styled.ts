import styled from "@emotion/styled";

export const Container = styled.div`
  width: 100%;
  max-width: 1440px;
  padding: 1rem 5rem;
  margin: 0 auto;
`;
