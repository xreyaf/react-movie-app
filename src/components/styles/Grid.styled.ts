import styled from "@emotion/styled";
import { motion } from "framer-motion";

export const GridStyled = styled(motion.div)`
  display: grid;
  grid-template-columns: repeat(4, minmax(0, 1fr));
  row-gap: 1.4rem;
  column-gap: 1.4rem;
  align-items: start;
  @media screen and (max-width: 1199.98px) {
    grid-template-columns: repeat(3, minmax(0, 1fr));
    transition: all .5s;
  }
  @media screen and (max-width: 991.98px) {
    row-gap: .8rem;
    column-gap: .8rem;
    transition: all .5s;
  }
  @media screen and (max-width: 816.98px) {
    grid-template-columns: repeat(2, minmax(0, 1fr));
    row-gap: .8rem;
    column-gap: .8rem;
    transition: all .5s;
  }
  @media screen and (max-width: 575.98px) {
    grid-template-columns: repeat(1, minmax(0, 1fr));
    transition: all .5s;
  }
`;
