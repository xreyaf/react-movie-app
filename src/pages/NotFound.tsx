import React from 'react';
import { Link } from 'react-router-dom';
import {
  ContainerStyled,
  NotFoundContainerStyled,
} from '../components/styles/Container.styled';
import { ButtonStyled } from '../components/styles/Button.styled';
import nf from '../assets/svg/404.svg';
import theme from '../styles/theme';

function NotFound() {
  return (
    <ContainerStyled>
      <NotFoundContainerStyled>
        <img
          style={{
            width: '70%',
            marginBottom: '2rem',
            height: 'auto',
          }}
          src={nf}
          alt="Not found"
        />
        <h2>Lost your way?</h2>
        <p>
          Oops! This is awkward. You are looking for something that doesn&apos;t
          actually exist.
        </p>
        <Link to="/">
          <ButtonStyled
            whileHover={{
              backgroundColor: theme.colors.primary500,
              cursor: 'pointer',
            }}
          >
            Go Home
          </ButtonStyled>
        </Link>
      </NotFoundContainerStyled>
    </ContainerStyled>
  );
}

export default NotFound;
