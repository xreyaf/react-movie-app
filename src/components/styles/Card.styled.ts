import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const CardStyled = styled(motion.div)`
  position: relative;
  border-radius: 1rem;
  overflow: hidden;
  background-color: ${theme.colors.grey900_90};
  width: 100%;
  min-height: 100%;
`;
export const CardImageWrapper = styled.div`
  display: inline-block;
  border-radius: 1rem;
  width: 100%;
  height: 88%;
  overflow: hidden;
`;

export const ImageStyled = styled(motion.img)`
  object-fit: cover;
  width: 100%;
  min-height: 100%;
  vertical-align: middle;
`;

export const CardInfo = styled.div`
  padding: .5rem;
  margin-bottom: .5rem;
  overflow: hidden;

  a {
    color: ${theme.colors.grey50};
    font-weight: 600;

    &:hover {
      color: ${theme.colors.white100};
      transition: color .3s;
    }
  }
`;