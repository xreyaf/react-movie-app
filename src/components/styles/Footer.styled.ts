import styled from "@emotion/styled";
import { theme } from "../../styles/theme";

export const FooterStyled = styled.footer`
  background: ${theme.colors.grey900_80};
  backdrop-filter: blur(2px);
  padding: 5rem 0;
  text-align: center;
`;
