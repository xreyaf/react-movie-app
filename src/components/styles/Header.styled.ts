import styled from "@emotion/styled";
import { theme } from "../../styles/theme";
import { motion } from "framer-motion";

export const HeaderStyled = styled.header`
  position: fixed;
width: 100%;
height: 10%;
  top: 0;
  left: 0;


  z-index: 100;
  background: ${theme.colors.grey900_80};
  backdrop-filter: blur(24px);

  

`;

export const Nav = styled.nav`
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 3rem;

  ul {
    display: flex;
    flex-direction: row;
    justify-content: end;
    align-items: center;
  }
`;

export const Logo = styled(motion.img)`
  max-width: 3rem;
`;

export const StyledLinkMenu = styled(motion.a)`
  display: inline-flex;
  gap: 0.5rem;
  align-items: center;
  padding: 1rem 1rem;
  font-weight: 700;
  font-size: 1.2rem;
  cursor: pointer;
  color: ${theme.colors.grey400};

  &:hover {
    color: ${theme.colors.primary400};
  }
`;
