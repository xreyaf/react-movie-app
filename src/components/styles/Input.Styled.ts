import styled from "@emotion/styled";
import { theme } from "../../styles/theme";

export const InputStyled = styled.div`
  position: relative;
  border-radius: 1rem;
  margin-bottom: 3rem;

  label {
    position: absolute;
    top: 2rem;
    left: 4rem;
    font-size: 1.5rem;
    color: ${theme.colors.grey600};
    cursor: text;
    user-select: none;
    transition: all .3s cubic-bezier(0.165, 0.84, 0.44, 1);

  }
`;

export const InputContent = styled.input`
  background: transparent;
  background: ${theme.colors.black10};
  z-index: 3;
  width: 100%;
  height: 5rem;
  padding: 2rem 1.3rem 1rem 3.9rem;
  font-size: 1.3rem;
  border-radius: 1rem;
  color: ${theme.colors.grey400};
  border: 3px solid ${theme.colors.grey800};
  outline: none;

  &:focus {
    border-color: ${theme.colors.primary500};

    & + label {
      transform: translateY(-1.2rem);
      font-size: 1rem;
    }

    &::placeholder {
      opacity: 0.5;
      transition: all .2s
    }

  }

  &::placeholder {
    opacity: 0;
  }

  &:not(:placeholder-shown) + label {
    transform: translateY(-1.2rem);
    font-size: 1rem;
  }
`;

export const IconWrapper = styled.div`
  display: inline-block;
  position: absolute;
  top: 2rem;
  left: 1.5rem;
  color: ${theme.colors.grey600};
`;