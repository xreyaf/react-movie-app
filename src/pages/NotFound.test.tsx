import { theme } from '../styles/theme';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@emotion/react';
import NotFound from './NotFound';
import { MemoryRouter } from 'react-router-dom';

test('it works', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>
    </ThemeProvider>
  );
  expect(container.firstChild).toMatchSnapshot();
});
