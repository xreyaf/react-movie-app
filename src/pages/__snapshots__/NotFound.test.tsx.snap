// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.emotion-0 {
  width: 100%;
  max-width: 1440px;
  padding: 1rem 5rem;
  margin: 0 auto 2rem auto;
  height: 100%;
}

@media screen and (max-width: 1199.98px) {
  .emotion-0 {
    padding: 1rem 2rem;
  }
}

@media screen and (max-width: 991.98px) {
  .emotion-0 {
    padding: 1rem 1.5rem;
  }
}

@media screen and (max-width: 816.98px) {
  .emotion-0 {
    padding: 1rem 1.2rem;
  }
}

@media screen and (max-width: 575.98px) {
  .emotion-0 {
    padding: 1rem 1rem;
  }
}

.emotion-1 {
  min-height: calc(100vh - 184px);
  max-width: 880px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  margin: 0 auto;
}

.emotion-1 h2 {
  margin-bottom: 2rem;
}

.emotion-1 p {
  margin-bottom: 2rem;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 0.5rem;
  outline: none;
  border: none;
  border-radius: 1rem;
  background: hsla(246, 88.3%, 69.8%, 1);
  color: hsla(0, 0%, 100%, 1);
  padding: 1rem 2rem;
  font-size: 1rem;
  margin-bottom: 0.6rem;
}

.emotion-2 svg {
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
}

<div
  class="emotion-0"
>
  <div
    class="emotion-1"
  >
    <img
      alt="Not found"
      src="test-file-stub"
      style="width: 70%; margin-bottom: 2rem; height: auto;"
    />
    <h2>
      Lost your way?
    </h2>
    <p>
      Oops! This is awkward. You are looking for something that doesn't actually exist.
    </p>
    <a
      href="/"
    >
      <button
        class="emotion-2"
      >
        Go Home
      </button>
    </a>
  </div>
</div>
`;
